### ISSUE TITLE
async R process invocations with large command lines leave stale processes

### ISSUE THREAD

**Issue by kevinushey** [rstudio/rstudio#3414]
### System details

    RStudio Edition : Desktop
    RStudio Version : v1.2.945
    OS Version      : macOS 10.13.6
    R Version       : 3.5.1

### Steps to reproduce the problem

Try launching an async R process with a large expression to be invoked, e.g.

```R
scream <- paste(rep("A", 1E4), collapse = "")
.rs.runAsyncRProcess(
   scream,
   onStdout = function(output) writeLines(output),
   onStderr = function(output) writeLines(output),
   onCompleted = function(status) writeLines("onCompleted")
)
```

### Describe the problem in detail

The process launches, but never finishes. The overly-large `-e` command line parameter gets ignored, and then R launches itself as an interactive process, waiting for `stdin`.

```R
> scream <- paste(rep("A", 1E4), collapse = "")
> .rs.runAsyncRProcess(
+    scream,
+    onStdout = function(output) writeLines(output),
+    onStderr = function(output) writeLines(output),
+    onCompleted = function(status) writeLines("onCompleted")
+ )
[1] TRUE
WARNING: '-e AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
```

### Describe the behavior you expected

Overly-large command lines shouldn't cause an interactive R process to launch. (We should instead write these commands to file, and instruct R to execute from those files)

**Comment by kevinushey** [comment]
Note that this problem has occurred in the wild -- for example, during deployments, the requested command line can become quite large especially when a large quantity of files are ignored or similar.

**Comment by jmcphers** [comment]
> during deployments, the requested command line can become quite large

This is very true, and we already do some temporary file writing to get around it (for example the list of files to deploy is now passed as a temp file instead of a long argument). Well worth insuring ourselves against these kinds of issues in general!

**Comment by jmcphers** [comment]
I don't think this warrants a fix for 1.2 at all at this point in the release. It's been an issue since day one and virtually no one has encountered it; for the very few that do, there are reasonable workarounds (structure your directories so that you aren't clicking Ignore on tons of individual files).

**Comment by kevinushey** [comment]
Sounds good -- I'll close this since effectively the issue will be resolved in v1.3 regardless.
